File: split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        char_set = list(set(s))
        sample = []
        count = 0
        for i in s:
            sample.append(i)
            if sample.count(char_set[0]) == sample.count(char_set[1]):
                count += 1
                sample.clear()
        return count 

File: leetcode/split-a-string-in-balanced-strings/split-a-string-in-balanced-strings.cpp
#include<stack>

class Solution {
public:
    int balancedStringSplit(string s) {
        stack<char> stack;
        int count = 0;
        
        for (int i = 0; i < s.length(); i++)
            if (stack.empty()){
                stack.push(s[i]);
                count++;
            }
            else if (stack.top() == s[i]){
                stack.push(s[i]);
            }
            else if (stack.top() != s[i]){
                stack.pop();
            }
        return count;
    }
};